extensions/amp-viqeo-player/0.1/amp-viqeo-player.js

```diff
@@ -0,0 +1,347 @@
+        (previewUrl
+            && previewUrl.length && decodeURI(previewUrl)
+        )
+        || (this.kindIsProd_ ? 'https://cdn.viqeo.tv/embed' : 'https://stage.embed.viqeo.tv');
+
+    // Create preview iframe source path
+    previewUrl = assertAbsoluteHttpOrHttpsUrl(
+        `${previewUrl}/?vid=${this.videoId_}&amp=true`);
+
+    const jsonParams = {
+      scriptPlayerInit: encodeURIComponent(scriptPlayerInit),
+      previewUrl: encodeURIComponent(previewUrl),
+      videoId: this.videoId_,
+      profileId: this.profileId_,
+      markTagParams: this.markTagParams_,
+    };
+
+    const iframe = getIframe(this.win, this.element, 'viqeoplayer', jsonParams);
+    this.unlistenMessage_ = listen(
+        this.win,
+        'message',
+        this.handleViqeoMessages_.bind(this)
+    );
+
+    return this.mutateElement(() => {
+      this.element.appendChild(iframe);
+      this.iframe_ = iframe;
+      this.applyFillContent(iframe);
+    }).then(() => {
+      return this.playerReadyPromise_;
+    });
+  }
+
+  /**
+   * @param {!Event|{data: !JsonObject}} event
+   * @return {?JsonObject|string|undefined}
+   * @private
+   * */
+  handleViqeoMessages_(event) {
+    const eventData = getData(event);
+    if (!eventData ||
+        event.source !== (this.iframe_ && this.iframe_.contentWindow) ||
+        eventData['source'] !== 'ViqeoPlayer') {
+      return;
+    }
+
+    const action = eventData['action'];
+    if (action === 'ready') {
+      this.element.dispatchCustomEvent(VideoEvents.LOAD);
+      this.playerReadyResolver_(this.iframe_);
+    } else if (action === 'play') {
+      this.element.dispatchCustomEvent(VideoEvents.PLAYING);
+    } else if (action === 'pause') {
+      this.element.dispatchCustomEvent(VideoEvents.PAUSE);
+    } else if (action === 'mute') {
+      this.element.dispatchCustomEvent(VideoEvents.MUTED);
+    } else if (action === 'unmute') {
+      this.element.dispatchCustomEvent(VideoEvents.UNMUTED);
+    } else if (action === 'volume') {
+      this.volume_ = parseFloat(eventData['value']);
+      if (this.volume_ === 0) {
+        this.element.dispatchCustomEvent(VideoEvents.MUTED);
+      } else {
+        this.element.dispatchCustomEvent(VideoEvents.UNMUTED);
+      }
+    }
+  }
+
+  /** @override */
+  unlayoutCallback() {
+    if (this.iframe_) {
+      removeElement(this.iframe_);
+      this.iframe_ = null;
+    }
+    if (this.unlistenMessage_) {
+      this.unlistenMessage_();
+    }
+
+    const deferred = new Deferred();
+    this.playerReadyPromise_ = deferred.promise;
+    this.playerReadyResolver_ = deferred.resolve;
+    return true; // Call layoutCallback again.
+  }
+
+  /** @override */
+  supportsPlatform() {
+    return true;
+  }
+
+  /** @override */
+  isInteractive() {
+    return true;
+  }
+
+  /** @override */
+  play() {
+    this.sendCommand_('play');
+  }
+
+  /** @override */
+  pause() {
+    this.sendCommand_('pause');
+  }
+
+  /** @override */
+  mute() {
```

Comment:
based on my testing `mute` and `unmuted` do not work.

Reply:
fixed